addons/web/static/src/core/l10n/time.js

```diff
@@ -0,0 +1,280 @@
+     * }?} params
+     */
+    constructor({ hour = 0, minute = 0, second = 0 } = {}) {
+        /**@type {number} */
+        this.hour = hour;
+        /**@type {number} */
+        this.minute = minute;
+        /**@type {number} */
+        this.second = second;
+
+        /**
+         * @private
+         * @type {boolean}
+         */
+        this._is24HourFormat = is24HourFormat();
+
+        /**
+         * @private
+         * @type {boolean}
+         */
+        this._isMeridiemFormat = isMeridiemFormat();
+    }
+
+    /**
+     * @param {number} rounding
+     */
+    roundMinutes(rounding) {
+        this.minute = Math.round(this.minute / rounding) * rounding;
+    }
+
+    /**
+     * @param {Time} time
+     * @returns {Time}
+     */
+    copy() {
+        return new Time(this);
+    }
+
+    /**
+     * @param {Time} other
+     * @param {boolean} [checkSeconds=false]
+     * @returns {boolean}
+     */
+    equals(other, checkSeconds = false) {
+        return other && this.hour === other.hour && this.minute === other.minute && (!checkSeconds || this.second === other.second);
+    }
+
+    /**
+     * Returns the formatted value of the time, with 24 of 12 hours
+     * format and with or without meridiems depending on the current
+     * localization time format.
+     *
+     * @param {boolean} [showSeconds=false]
+     * @returns {string}
+     */
+    toString(showSeconds = false) {
+        const hourFormat = this._is24HourFormat ? "H" : "h";
+        const secondFormat = showSeconds ? ":ss" : "";
+        const meridiemFormat = this._isMeridiemFormat ? "a" : "";
+        return this.toDateTime().toFormat(`${hourFormat}:mm${secondFormat}${meridiemFormat}`).toLowerCase();
+    }
+
+    /**
+     * @returns {DateTime}
+     */
+    toDateTime() {
+        return DateTime.fromObject({
+            hour: this.hour,
+            minute: this.minute,
+            second: this.second,
+        });
+    }
+}
+
+/**
+ * Returns whether the given format is a 24-hour format.
+ * Falls back to localization time format if none is given.
+ *
+ * @param {string} format
+ */
+export function is24HourFormat(format) {
+    return /H/.test(format || localization.timeFormat);
+}
+
+/**
+ * Returns whether the given format uses a meridiem suffix (AM/PM).
+ * Falls back to localization time format if none is given.
+ *
+ * @param {string} format
+ */
+export function isMeridiemFormat(format) {
+    return /a/.test(format || localization.timeFormat);
+}
+
+/**
+ * Tries to parse a Time object from a time string
+ * representation such as:
+ * "10:15"  -> 10:15:00
+ * "2h5"    -> 02:50:00
+ * "1015"   -> 10:15:00
+ * "125"    -> 12:50:00
+ * "315"    -> 03:15:00
+ * "5:15pm" -> 17:15:00
+ *
+ * Returns null if the value could not be parsed.
+ *
+ * @param {string} value
+ * @param {boolean} [parseSeconds]
+ * @returns {Time}
+ */
+export function parseTime(value, parseSeconds) {
+    const { isPm, isAm } = meridiemCheck(value);
```

Comment:
I would add the same sanity check for null values existing in `parseDate` & `parseDateTime`
```suggestion
    if (!value) {
        return null;
    }
    const { isPm, isAm } = meridiemCheck(value);
```